pkg/pillar/cas/containerd.go

```diff
@@ -705,6 +708,85 @@ func (c *containerdCAS) PrepareContainerRootDir(rootPath, reference, rootBlobSha
 		logrus.Errorf(err.Error())
 		return err
 	}
+	// On kubevirt eve we also write the mountpoints and cmdline files to the rootPath.
+	// Once rootPath is populated with all necessary files, this rootPath directory will be
+	// converted to a PVC and passed in to domainmgr to attach to VM as rootdisk
+	// NOTE: For kvm eve these files are generated and passed to bootloader in pkg/pillar/containerd/oci.go:AddLoader()
+	if base.IsHVTypeKube() {
+
+		// Mount the snapshot
+		if err := c.MountSnapshot(snapshotID, GetRoofFsPath(rootPath)); err != nil {
```

Comment:
You leave the mount in place once this is done.
Does that mean there is a contract with some code which runs later where it assumes it is already mounted? That might be a reason to move this code, or if not document this contract in comments.

Reply:
Yes we need this mount to exist so that this container directory will be converted to qcow2->PVC. But the unmount happens after qcow2 creation in pkg/pillar/volumehandlers/csihandler.go.

Sure I will add  comments in here.